pkg/ccip/chainaccessor/event.go

```diff
@@ -0,0 +1,42 @@
+package chainaccessor
+
+import (
+	"errors"
+
+	"github.com/smartcontractkit/chainlink-ccip/pkg/chainaccessor"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/chainlink-ton/pkg/ccip/bindings/onramp"
+)
+
+// convertCCIPMessageSent converts a TON CCIPMessageSent event to a generic chainaccessor.SendRequestedEvent event
+func (a *TONAccessor) convertCCIPMessageSent(
+	tonEvent *onramp.CCIPMessageSent,
+) (*chainaccessor.SendRequestedEvent, error) {
+	if a.chainSelector == 0 {
+		return nil, errors.New("source chain selector cannot be zero")
+	}
+
+	msg := ccipocr3.Message{
+		Header: ccipocr3.RampMessageHeader{
+			MessageID:           ccipocr3.Bytes32(tonEvent.Message.Header.MessageID),
+			SourceChainSelector: a.chainSelector,
+			DestChainSelector:   ccipocr3.ChainSelector(tonEvent.Message.Header.DestChainSelector),
+			SequenceNumber:      ccipocr3.SeqNum(tonEvent.Message.Header.SequenceNumber),
+			Nonce:               tonEvent.Message.Header.Nonce,
+		},
+		Sender:         ccipocr3.UnknownAddress(tonEvent.Message.Sender.String()),
+		Data:           ccipocr3.Bytes(tonEvent.Message.Body.Data),
+		Receiver:       ccipocr3.UnknownAddress(tonEvent.Message.Body.Receiver),
+		ExtraArgs:      tonEvent.Message.Body.ExtraArgs.ToBOC(),
```

Comment:
Hmm we need to consider how to handle extraArgs: it would be nice if we could convert it to a byte slice here to avoid needing TON libraries to parse it on the other end

However, if we want to verify the message ID the MessageHasher has to convert this back to a TON message anyway since we rely on TON's native hashing

\cc @huangzhen1997 do we have MessageHasher implemented anywhere?

Reply:
Just created a draft [PR](https://github.com/smartcontractkit/chainlink-ton/pull/128) for it, didn't build MsgHasher yet since it's hashing Any2TVM. For TVM2TON case we should be using evm message hasher